fuzz/oss-fuzz-build.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash -eu
+
+TELEPORT_PREFIX="github.com/gravitational/teleport"
+
+prepare_teleport() {
+
+  go mod tidy
+
+  go get github.com/AdamKorcz/go-118-fuzz-build/utils
+  go get k8s.io/client-go/tools/clientcmd@v0.23.3
+  go get github.com/ThalesIgnite/crypto11@v1.2.4
+
+  # Fix /root/go/pkg/mod/github.com/aws/aws-sdk-go-v2/internal/ini@v1.3.0/fuzz.go:13:21:
+  # not enough arguments in call to Parse
+  rm /root/go/pkg/mod/github.com/aws/aws-sdk-go-v2/internal/ini@*/fuzz.go
```

Comment:
Is this somehow related to this: https://github.com/google/oss-fuzz/blob/master/projects/teleport/build.sh#L19 ?

Reply:
[go.mod](https://github.com/gravitational/teleport/blob/b2b2312a5e9b790070b1bb2d5ef05110198c23de/go.mod#L158) we uses `github.com/aws/aws-sdk-go-v2/internal/ini` with the version 1.3.0.

This specific version uses an obsolete fuzzer https://github.com/aws/aws-sdk-go-v2/blob/v1.3.0/internal/ini/fuzz.go which was later in 1.16.5 removed [here](https://github.com/aws/aws-sdk-go-v2/commit/a34da05fbb2c27569c66592c7d489f55acf7bd94). The problem is that fuzzer calls

```go
if _, err := Parse(b); err != nil {
```

but the signature in 1.3.0 is already [different](https://github.com/aws/aws-sdk-go-v2/blob/a34da05fbb2c27569c66592c7d489f55acf7bd94/internal/ini/ini.go):

```go
func Parse(f io.Reader, path string) (Sections, error) {
```

This does not trigger an error during normal builds, but I did some debugging and during the compilation of the target when using `go-118-fuzz-build` (dependency for oss-fuzz), it links against all files, including the useless `fuzz.go`:

```
$ strace -s 4096 -f -e execve go-118-fuzz-build -o $fuzzer.a -func $function github.com/gravitational/teleport/lib/services 2>&1
[ .. SNIP .. ]

[pid  6349] execve("/root/.go/pkg/tool/linux_amd64/compile", ["/root/.go/pkg/tool/linux_amd64/compile", "-o", "/tmp/go-build1849907098/b451/_pkg_.a", "-trimpath", "/root/go/pkg/mod/github.com/aws/aws-sdk-go-v2/internal/ini@v1.3.0=>github.com/aws/aws-sdk-go-v2/internal/ini@v1.3.0;/tmp/go-build1849907098/b451=>", "-p", "github.com/aws/aws-sdk-go-v2/internal/ini", "-lang=go1.15", "-complete", "-installsuffix", "shared", "-buildid", "_Y4HsbPwFpnPCDjtZue_/_Y4HsbPwFpnPCDjtZue_", "-goversion", "go1.18", "-shared", "-d=libfuzzer", "-nolocalimports", "-importcfg", "/tmp/go-build1849907098/b451/importcfg", "-pack", 
[ .. SNIP ..] 
"/root/go/pkg/mod/github.com/aws/aws-sdk-go-v2/internal/ini@v1.3.0/errors.go", "/root/go/pkg/mod/github.com/aws/aws-sdk-go-v2/internal/ini@v1.3.0/expression.go", "/root/go/pkg/mod/github.com/aws/aws-sdk-go-v2/internal/ini@v1.3.0/fuzz.go", "/root/go/pkg/mod/github.com/aws/aws-sdk-go-v2/internal/ini@v1.3.0/go_module_metadata.go", 
[ .. SNIP ..] 
```

I know this is far from ideal, but if we keep this dependency version, I am not sure what else to do here than remove the file, since the other solutions involve modifying the already established compilation process.